### PR TITLE
Add minor clarifying documentation to AABBQuery

### PR DESCRIPTION
**Pull request by klarh**
## Description
This adds slightly more context to the AABBQuery docstring.

## Motivation and Context
Previously, the docstring for AABBQuery wasn't very helpful. This changeset defines what "AABB" means and adds the constructor arguments.

## How Has This Been Tested?

It hasn't, but it is only a documentation change.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [x] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

**Comment by klarh** [comment]
Let me know if you want me to re-cythonize the branch to actually update the docstring.

File: freud/locality.pyx [hunk 1]
```diff
@@ -774,9 +774,15 @@ cdef class _RawPoints(NeighborQuery):
 
 
 cdef class AABBQuery(NeighborQuery):
-    R"""Use an AABB tree to find neighbors.
+    R"""Use an Axis-Aligned Bounding Box (AABB) tree to find neighbors.
```
Comment (vyasr) [inline freud/locality.pyx]: One thing we need to be a bit better about in freud is citations. We improved on this quite a bit when revising for 2.0, but here we are thoroughly lacking. Would you mind adding a citation of Michael Howard's paper https://www.sciencedirect.com/science/article/pii/S0010465516300182 to the bibtex file (in doc/source/reference/freud.bib) and then referencing it here? The format for inline citations that we're using now looks like :cite:`vanAnders:2014aa` (from e.g. pmft.pyx).

File: freud/locality.pyx [hunk 2]
```diff
@@ -774,9 +774,15 @@ cdef class _RawPoints(NeighborQuery):
...
 
     Also available as ``freud.AABBQuery``.
+
+    Args:
+        box (:class:`freud.box.Box`):
+            Simulation box.
+        points (:class:`np.ndarray`):
```
Comment (vyasr) [inline freud/locality.pyx]: Could you copy this line in docstring from `NeighborQuery`, which also indicates the appropriate shape of the points array?

Comment (vyasr) [reply freud/locality.pyx]: It would also be great if you could propagate this change to `LinkCell`.

**Review by vyasr — CHANGES_REQUESTED** [review]
I have a couple of minor change requests. Let me know if you won't have time, I could make the changes myself too.

**Review by vyasr — APPROVED** [review]
Thanks!

**Pull request merged**